doc/api/n-api.md

```diff
@@ -2267,6 +2267,7 @@ typedef enum {
   napi_float64_array,
   napi_bigint64_array,
   napi_biguint64_array,
+  napi_float16_array,
```

Comment:
Without breaking the ABI, we still need to mark this as NAPI_EXPERIMENTAL as mentioned in https://github.com/nodejs/node/pull/58879#discussion_r2173842424 until this is been backported to all LTS lines.

In addition, we need a feature macro, to allow addons to detect a new feature: https://github.com/nodejs/node/blob/main/doc/contributing/adding-new-napi-api.md#process-and-approval.

Reply:
I'm not saying this breaks ABI. My point is we need guard this with `NAPI_EXPERIMENTAL` and a feature macro to allow addons detect that APIs like `napi_create_typedarray` are supported with `napi_float16_array` at both build time and run time.